### DISCUSSION TITLE
Import error: Cannot import name 'Exit'

### DISCUSSION THREAD

**Discussion by mariaubergine** [explosion/spaCy#9457] [Help: Coding & Implementations] [answered]
Hi there, 

We have developed a basic NLP pipeline using Spacy 3 to detect different entities. For now, we're using a largely rule-based approach (with EntityRulers) on top of en_core_web_lg models and intend to scale with statistical learning going forward. In development, we're using spacy version 3.0.7, and the saved models are compatible.

Checking the version of the spacy model using spacy.info yields:

```
'name': 'core_web_lg',
 'version': '3.0.0',
 'spacy_version': '>=3.0.0,<3.1.0'
```

We're using k9s and Airflow to deploy our model, but am facing a strange import error: `cannot import name 'Exit'` - I've looked on stack overflow and Github with no signs of a similar error faced by other people. We didn't use to face this error with spacy V2 but only started to get this error after we upgraded to spacy 3.0.7. The error is thrown up for all scripts that attempt to import spacy. Has anyone come across this before? If so, do you have any tips on how to de-bug and resolve this error?

**Comment by ljvmiranda921** [comment, marked as answer]
Hi @maria-tan-abv , is it possible to see the full traceback? The traceback should show which file and specific line of code causes the error. We can probably start from there 👍

**Reply by mariaubergine** [reply]
Hi @ljvmiranda921 thanks for your help! This is the full traceback from Airflow. 

```
File "/opt/bitnami/airflow/dags/git-abv-dags/daily_label_update.py", line 38, in <module>
    import spacy
  File "/opt/bitnami/airflow/venv/lib/python3.6/site-packages/spacy/__init__.py", line 14, in <module>
    from .cli.info import info  # noqa: F401
  File "/opt/bitnami/airflow/venv/lib/python3.6/site-packages/spacy/cli/__init__.py", line 3, in <module>
    from ._util import app, setup_cli  # noqa: F401
  File "/opt/bitnami/airflow/venv/lib/python3.6/site-packages/spacy/cli/_util.py", line 8, in <module>
    import typer
  File "/opt/bitnami/airflow/venv/lib/python3.6/site-packages/typer/__init__.py", line 7, in <module>
    from click.exceptions import Exit as Exit
ImportError: cannot import name 'Exit'
```

**Reply by ljvmiranda921** [reply]
Hi @maria-tan-abv ! Hmm, my best guess is that there are mismatched versions between typer and click. I was able to reproduce the error by installing `spacy==3.0.7`, `typer==0.3.2`, and `click<7.0.0`. The probable culprit is click, [which should be `>=7.1.1, 7.2.0`](https://github.com/tiangolo/typer/blob/0.3.2/pyproject.toml). Try upgrading its version and see if it works.

Another way is to completely uninstall spaCy and reinstall in a fresh environment--instead of fixing sub-dependencies. You can then pin the dependencies later on via `pip freeze` or tools like [pip-tools](https://github.com/jazzband/pip-tools) or [poetry](https://github.com/python-poetry/poetry).

**Reply by mariaubergine** [reply]
Hi @ljvmiranda921 thank you so much! This worked!
